nservicebus/upgrades/5to6-moving-away-from-ibus.md

```diff
@@ -0,0 +1,84 @@
+
+The `IBus` interface was one of the key interfaces when using previous versions of NServiceBus. It provided access to many of the important operations like sending messages, subscribing to events, and manipulating headers. The `IBus` interface was injected into the container using dependency injection and was accessible in all the message handlers and other injected components in the container as well. 
+
+There are several scenarios when using IBus:
+
+## During Endpoint Creation
+
+In previous versions of NServiceBus to start a new instance of an endpoint, either the `Bus` static class or the `IStartableBus` interface was used. In Version 6 these two concepts have been replaced. More information can be found in the [hosting](/nservicebus/hosting/) section of the documentation.
+
+`BusConfiguration` has been deprecated. Use `EndpointConfiguration` instead to configure and start the endpoint. 
+
+Instead of the `Bus` class, a new class called `Endpoint` helps to start the endpoint.
+
+snippet: 5to6-endpoint-start-stop
+
+Starting the endpoint, provides access to `IEndpointInstance` which can be used to send messages during endpoint start up instead.
+
+## Sending messages outside message handlers
+
+A common use of `IBus` is to invoke bus operations outside of message handlers, such as sending a message from an ASP.NET request or from a UI input. Instead of an `IBus` the `IEndpointInstance` offers all available bus operations outside the message processing pipeline. For example:
+
+If the endpoint is hosted using NServiceBus.Host, use the [IWantToRunWhenEndpointStartsAndStops interface](/nservicebus/upgrades/host-6to7.md) as outlined in the NServiceBus.Host documentation.
+
+## Sending messages inside message handlers
+
+The public `IBus` injection property in message handlers can now be safely deleted.
+
+The message handlers now includes an additional `IMessageHandlerContext` parameter, which provides the methods that used to be called from `IBus`. Use the `IMessageHandlerContext` to send and publish messages from within the message handler. 
+
+snippet:5to6-bus-send-publish
+
+
+## Accessing the CurrentMessageContext
+
+Previously it was possible to access message parameters such `MessageId`, `ReplyToAddress` and the message headers via the `CurrentMessageContext` property on the `IBus` interface. These message properties are now available directly via the message handler context parameter.
+
+snippet: 5to6-messagecontext
+
+
+## Dependency Injection
+
+In previous versions of NServiceBus the `IBus` interface was automatically registered in the IoC container. In Version 6, the new context aware interfaces namely the `IEndpointInstance`, `IMessageSession` or `IMessageHandlerContext` will not be automatically registered in the IoC container. 
+
+The `IEndpointInstance` interface or the `IMessageSession` interface when using the NServiceBus.Host can be injected into the container.
```

Comment:
I don't think `IEndpointInstance` can be injected manually into container when NSB.Host is used...